go/vt/sqlparser/sql.y

```diff
@@ -410,15 +418,38 @@ id_or_var:
   {
     $$ = NewColIdentWithAt(string($1), NoAt)
   }
-| AT_ID
+| at_id
   {
     $$ = NewColIdentWithAt(string($1), SingleAt)
   }
-| AT_AT_ID
+| at_at_id
   {
     $$ = NewColIdentWithAt(string($1), DoubleAt)
   }
 
+at_at_id:
+  at_allow_carrats '@' ID
+  {
+    $$ = $3
+    setAllowCarrats(yylex,false)
+    setAllowBlanks(yylex,true)
+  }
+
+at_id:
+  at_allow_carrats ID
+  {
+    $$ = $2
+    setAllowCarrats(yylex,false)
+    setAllowBlanks(yylex,true)
+  }
+
+at_allow_carrats:
+  '@'
+  {
+    setAllowCarrats(yylex,true)
+    setAllowBlanks(yylex,false)
```

Comment:
I have one final question. Since this is LALR(1), there are situations where the parser has the right to look ahead a token before reducing the LHS. There was one situation that I encountered before where the parser had already looked ahead before deciding to execute the code.

Is there a way to be sure that the parser won't look ahead until it's done executing this code snippet?

Reply:
@GuptaManan100 I think for all the cases (`at_at_id`, `at_id`, `at_allow_carrats`) we do not need to allow blanks. Do you know of a case where space is allowed?